core/vm/logger.go

```diff
@@ -18,12 +18,157 @@ package vm
 
 import (
 	"fmt"
+	"math/big"
 	"os"
 	"unicode"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
+type Storage map[common.Hash]common.Hash
+
+func (self Storage) Copy() Storage {
+	cpy := make(Storage)
+	for key, value := range self {
+		cpy[key] = value
+	}
+
+	return cpy
+}
+
+// StructuredLogCollector is the basic interface to capture emited logs by the EVM logger.
+type StructuredLogCollector interface {
+	// Adds the structured log to the collector.
+	AddStructuredLog(StructLog)
+}
+
+// LogCfg are the configuration options for structured logger the EVM
+type LogCfg struct {
+	DisableMemory  bool                   // disable memory capture
+	DisableStack   bool                   // disable stack capture
+	DisableStorage bool                   // disable storage capture
+	FullStorage    bool                   // show full storage (slow)
+	Collector      StructuredLogCollector // the log collector
+}
```

Comment:
More naming nits (if only there was something else I could complain about...):
Why is Cfg abbreviated here but Config isn't?